python/events/src/hydrotools/events/models/interception.py

```diff
@@ -0,0 +1,376 @@
+        mm/hour, then canopy_storate should be in mm).
+    trunk_evaporation: float, required
+        Fraction of evaporation originating from vegetation "trunks." Typically
+        0.02 (Miralles et al, 2010). Notated $\\epsilon$. Unitless. Valid
+        range is [0.0, 1.0)
+    evaporation_rate: float, required
+        Average evaporation rate. Typically 0.3 mm/hour (Miralles et al, 2010).
+        Notated $E_c$. Expected to vary little between locations
+        (± 0.1 mm/hour, standard deviation). Should be in the same measurement
+        units as rainfall_rate.
+
+    Returns
+    -------
+    Maximum canopy storage (L) as a float.
+    """
+    # Validate
+    assert rainfall_rate > 0, "rainfall rate must be a number > 0"
+    assert canopy_storage > 0, "canopy storage must be a number > 0"
+    assert evaporation_rate > 0, "evaporation rate must be a number > 0"
+    assert 0.0 <= trunk_evaporation < 1.0, \
+        "trunk evaporation range is [0.0, 1.0)"
+    if evaporation_rate >= rainfall_rate:
+        message = ("evaporation rate >= rainfall rate, " +
+            "use result with caution")
+        warnings.warn(message, UserWarning)
+
+    return (-((rainfall_rate * canopy_storage) /
+        ((1-trunk_evaporation) * evaporation_rate)) *
+        np.log(1-((1-trunk_evaporation)*evaporation_rate/rainfall_rate)))
+
+def compute_trunk_saturation(
+        rainfall_rate: float,
+        canopy_fraction: float,
+        canopy_saturation: float,
+        evaporation_rate: float,
+        trunk_capacity: float,
+        trunk_fraction: float,
+        trunk_evaporation: float
+) -> float:
+    """
+    Compute the amount of rainfall necessary to saturate the trunks under
+    the canopy. Notated as $P_g''$ in Miralles et al, 2010. Typically in mm.
+
+    Parameters
+    ----------
+    rainfall_rate: float, required
+        Mean rainfall rate in L/T, typically mm/hour. Notated 'R'.
+    canopy_fraction: float, required
+        Projected portion of canopy cover, unitless. Notated 'c'. Valid
+        range is (0.0, 1.0]
+    canopy_saturation: float, required
+        Maximum canopy storage, typically mm. Notated $P_g'$. Can be computed
+        using `compute_canopy_saturation`.
+    evaporation_rate: float, required
+        Average evaporation rate. Typically 0.3 mm/hour (Miralles et al, 2010).
+        Notated $E_c$. Expected to vary little between locations
+        (± 0.1 mm/hour, standard deviation). Should be in the same measurement
+        units as rainfall_rate.
+    trunk_capacity: float, required
+        Minimum depth of water required to saturate trunks under the canopy.
+        Typically 0.02 mm (Miralles et al, 2010). Notated $S_t$. Measurement
+        units should correspond to rainfall_rate (i.e. if rainfall_rate is in
+        mm/hour, then trunk_capacity should be in mm).
+    trunk_fraction: float, required
+        Fraction of rainfall hitting the "trunks" under the canopy. Typically
```

Comment:
These unitless parameters with "typical" values may be prime candidates for optional/default params